views/webs/WebsView.js

```diff
@@ -0,0 +1,1190 @@
+
+        return `B. ${date ? `<strong>${date}</strong>` : "[date unknown]"} ${
+            place ? `in ${place}` : "[location unknown]"
+        }.`;
+    }
+
+    /**
+     * Generate text that display when and where the person died
+     */
+    function deathString(person) {
+        var string = "",
+            date = humanDate(person.getDeathDate()),
+            place = person.getDeathLocation();
+
+        return `D. ${date ? `<strong>${date}</strong>` : "[date unknown]"} ${
+            place ? `in ${place}` : "[location unknown]"
+        }.`;
+    }
+
+    var monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+
+    /**
+     * Turn a wikitree formatted date into a humanreadable date
+     */
+    function humanDate(dateString) {
+        if (dateString && /\d{4}-\d{2}-\d{2}/.test(dateString)) {
+            var parts = dateString.split("-"),
+                year = parseInt(parts[0], 10),
+                month = parseInt(parts[1], 10),
+                day = parseInt(parts[2], 10);
+            if (year) {
+                if (month) {
+                    if (day) {
+                        return `${day} ${monthNames[month - 1]} ${year}`;
+                    } else {
+                        return `${monthNames[month - 1]} ${year}`;
+                    }
+                } else {
+                    return year;
+                }
+            }
+        } else {
+            return dateString;
+        }
+    }
+
+    /**
+     * Shorten the name if it will be too long to display in full.
+     */
+    function getShortName(person) {
+        const maxLength = 20;
+
+        const birthName = person.getDisplayName();
+        const middleInitialName = `${person._data.FirstName} ${person._data.MiddleInitial} ${person._data.LastNameAtBirth}`;
+        const noMiddleInitialName = `${person._data.FirstName} ${person._data.LastNameAtBirth}`;
+
+        if (birthName.length < maxLength) {
+            return birthName;
+        } else if (middleInitialName.length < maxLength) {
+            return middleInitialName;
+        } else if (noMiddleInitialName.length < maxLength) {
+            return noMiddleInitialName;
+        } else {
+            return `${person._data.FirstName.substring(0, 1)}. ${person._data.LastNameAtBirth}`;
+        }
+    }
+    /**
+     * Get the first initial of the first name only
+     */
+    function getFirstInitial(person) {
+        return `${person._data.FirstName.substring(0, 1)}`;
+
+    }
+    /**
+     * Get the full first name only
+     */
+    function getFirstName(person) {
+        return `${person._data.FirstName}`;
+    }
+
+    /**
+     * Sort by the birth year from earliest to latest.
+     */
+    function sortByBirthDate(list) {
+        list.sort((a, b) => {
+            const aBirthYear = a._data.BirthDate.split("-")[0];
+            const bBirthYear = b._data.BirthDate.split("-")[0];
```

Comment:
Date strings in format of `YYYY-MM-DD` are imho sortable even without parsing or splitting.